code/common/cloudutils.q

```diff
@@ -0,0 +1,59 @@
+/a function to execute system commands and return a log message depending on the resulting exit code:
+/used for both launch and killproc
+syscomm:{[params;cmd]
+    /params is (i) a dictionary if syscomm has been called by launch, and (ii) a single string if it has been called by killproc
+    /cmd is the command to be executed as a string
+    
+    prevexitcode:first "I"$system cmd,"; echo $?";
+ 
+    $[99h=type params;
+        [$[0=prevexitcode;
+            .lg.o[`launchprocess;"Successful execution: Starting ",params[`procname]];
+          1=prevexitcode;
+            .lg.e[`launchprocess;"Error: ",params[`procname]," failed to start"];
+          2=prevexitcode;
+            .lg.e[`launchprocess;"Error: ",params[`procname]," already running"];
+          .lg.e[`launchprocess;"Unknown error encountered"]]];
+        
+        [$[0=prevexitcode;
+            .lg.o[`killprocess;"Successful execution: Terminating ",params];
+          1=prevexitcode;
+            .lg.e[`killprocess; params," was not running"];
+          2=prevexitcode;
+            .lg.e[`killprocess;"Process name ",params," was not found"];
+          3=prevexitcode;
+            .lg.e[`killprocess;"Failed to terminate "params]
+          .lg.e[`killprocess;"Unknown error encountered"]]] 
+      ]
+   }
+
+/function which lets us call launchprocess.sh from inside a TorQ process
+/it takes a dictionary of parameters which will be passed to launchprocess.sh, i.e "-procname rdb1 -proctype rdb" etc.
+launch:{ [params]
+    /exit immediately if process name and type aren't provided
+    if[not 11b~`procname`proctype in key params;0N!
```

Comment:
I imagine the `0N!` on this line is a remnant from some debugging? Presumably this should be removed. Can also use `all` here - should do the same thing, think it's just a little more immediately clear

```suggestion
    if[not all `procname`proctype in key params;
```